web/app/routes/authenticated/dashboard.js

```diff
@@ -67,52 +67,12 @@ export default class DashboardRoute extends Route {
         return result.hits;
       });
 
-    // Get recently viewed docs from app data.
-    const recentlyViewedDocIDs = await this.recentDocs.get.perform();
-
-    // For each recently viewed doc (max 4 docs), fetch doc metadata from the
-    // app backend.
-    const recentlyViewedPromises = recentlyViewedDocIDs
-      .slice(0, 4)
-      .map((docID) =>
-        this.fetchSvc
-          .fetch("/api/v1/documents/" + docID)
-          .then((resp) => resp.json())
-          .catch((err) => {
-            console.log(
-              `Error getting recently updated document (${hit.objectID}):`,
-              err
-            );
-          })
-      );
-
-    // Create promise for all Algolia recently viewed docs promises.
-    const recentlyViewedDocsPromise = Promise.allSettled(
-      recentlyViewedPromises
-    );
-
-    // Create array of docs that we also indexed in Algolia.
-    // We can't display documents without this data.
-    const recentlyViewedDocs = recentlyViewedDocsPromise.then((promises) => {
-      let recentlyViewedDocs = [];
-
-      promises.forEach((promise, index) => {
-        if (promise.status == "fulfilled") {
-          let doc = promise.value;
-          doc.modifiedAgo = `Modified ${timeAgo(
-            new Date(doc.modifiedTime * 1000)
-          )}`;
-
-          recentlyViewedDocs.push(promise.value);
-        }
-      });
-
-      return recentlyViewedDocs;
-    });
+    if (this.recentDocs.all === null) {
+      await this.recentDocs.fetchAll.perform();
```

Comment:
Should we add some error checking here?

Reply:
What did you have in mind? Here I'm relying on `fetchAll`'s error handling

Reply:
It looks like `fetchAll` just logs and throws an error so it would end up here? I might be missing something though.

Reply:
Also this isn't necessarily a blocker, but I just wanted to make sure that the dashboard would still load if there was an error with `fetchAll` - did you test that?

Reply:
I see what you mean. Check out these changes: https://github.com/hashicorp-forge/hermes/pull/38/commits/a634c7d2d837e77620d2dc15cfa0632e9f87176e

We now let failed `fetchAll` through and show an error in the template with the option to refresh.

In this example it's hard-coded to fail, but it should give you an idea of the implementation:
https://user-images.githubusercontent.com/754957/227072262-c2cbc23e-28d7-4200-9186-2a3e8e4063cd.mp4

Reply:
Looks great! :shipit: